davinci/src/main/kotlin/com/pingidentity/davinci/collector/Form.kt

```diff
@@ -27,15 +28,17 @@ internal object Form {
      * This function takes a JSON object and extracts the "form" field. It then iterates over the "fields" array in the "components" object,
      * parsing each field into a collector and adding it to a list.
      *
+     * @param daVinci The DaVinci instance to be injected.
      * @param json The JSON object to parse.
      * @return A list of collectors parsed from the JSON object.
      */
     fun parse(
+        daVinci: DaVinci,
         json: JsonObject,
     ): Collectors {
         val collectors = mutableListOf<Collector<*>>()
         json["form"]?.jsonObject?.get("components")?.jsonObject?.get("fields")?.jsonArray?.let { array ->
```

Comment:
Static strings for keys ideally should be part of a static enum or file to centralize those.